File: app/src/api/quiz.ts
import { authFetch } from './client';

export interface QuizQuestion {
  dialogue_id: string;
  prompt: string;
  options: string[];
}

export interface QuizAnswer {
  dialogue_id: string;
  selected_text: string;
}

export interface QuizResult {
  correct: number;
  total: number;
  xp_awarded: number;
}

// کوئیز درک شنیداریِ یک صحنه — کاملاً از روی دیالوگ‌های واقعی همان صحنه
// (و صحنه‌های دیگر برای گزینه‌های غلط) روی بک‌اند ساخته می‌شود.
export async function fetchSceneQuiz(sceneId: string): Promise<QuizQuestion[]> {
  const res = await authFetch(`/v1/scenes/${sceneId}/quiz`, { method: 'GET' });
  if (!res.ok) throw new Error('failed to fetch quiz');
  const data = (await res.json()) as QuizQuestion[] | null;
  return data || [];
}

// نمره‌دهی سمت سرور انجام می‌شود (نه کلاینت) چون گزینه‌ها هر بار تصادفی‌اند.
export async function submitSceneQuiz(sceneId: string, answers: QuizAnswer[]): Promise<QuizResult> {
  const res = await authFetch(`/v1/scenes/${sceneId}/quiz/submit`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ answers }),
  });
  if (!res.ok) throw new Error('failed to submit quiz');
  return (await res.json()) as QuizResult;
}

File: internal/service/learning/dto/quiz.go
package dto

// QuizQuestion یک سوال چهارگزینه‌ای درک شنیداری است که کاملاً از روی
// دیالوگ‌های واقعیِ همین صحنه (و صحنه‌های دیگر برای گزینه‌های غلط) ساخته
// می‌شود — محتوای دستی جداگانه لازم ندارد، پس با هر صحنه‌ی جدیدی که ادمین
// می‌سازد خودکار کار می‌کند.
type QuizQuestion struct {
	DialogueID string `json:"dialogue_id"`
	// Prompt ترجمه‌ی فارسیِ همان جمله است؛ کاربر باید جمله‌ی انگلیسیِ
	// درست را از بین گزینه‌ها تشخیص بدهد.
	Prompt  string   `json:"prompt"`
	Options []string `json:"options"`
}

type QuizAnswer struct {
	DialogueID   string `json:"dialogue_id"`
	SelectedText string `json:"selected_text"`
}

type QuizSubmitRequest struct {
	Answers []QuizAnswer `json:"answers"`
}

type QuizResult struct {
	Correct   int `json:"correct"`
	Total     int `json:"total"`
	XPAwarded int `json:"xp_awarded"`
}
